modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/portlet/action/test/AutoSaveArticleMVCResourceCommandTest.java

```diff
@@ -0,0 +1,341 @@
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockMultipartHttpServletRequest;
+
+/**
+ * @author Mikel Lorza
+ */
+@RunWith(Arquillian.class)
+public class AutoSaveArticleMVCResourceCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testServeResource() throws Exception {
+		JSONObject jsonObject = _serveResource(
+			_getMockLiferayResourceRequest());
+
+		Assert.assertTrue(jsonObject.getBoolean("success"));
+
+		JournalArticle journalArticle =
+			_journalArticleLocalService.fetchArticle(
+				_group.getGroupId(), jsonObject.getString("articleId"));
+
+		Assert.assertNotNull(journalArticle);
+
+		Assert.assertNotNull("title", jsonObject.getString("friendlyURL"));
+
+		Date modifiedDate = journalArticle.getModifiedDate();
+
+		Assert.assertNotNull(modifiedDate);
+		Assert.assertEquals(
+			modifiedDate.getTime(), jsonObject.getLong("modifiedDate"));
+
+		Assert.assertEquals(
+			WorkflowConstants.STATUS_DRAFT, journalArticle.getStatus());
+
+		Assert.assertEquals("1", jsonObject.getString("version"));
+
+		jsonObject = _serveResource(
+			_getMockLiferayResourceRequest(
+				journalArticle.getArticleId(),
+				journalArticle.getFriendlyURLMap()));
+
+		Assert.assertTrue(jsonObject.getBoolean("success"));
+
+		journalArticle = _journalArticleLocalService.fetchArticle(
+			_group.getGroupId(), jsonObject.getString("articleId"));
+
+		Assert.assertFalse(jsonObject.has("friendlyURL"));
+
+		modifiedDate = journalArticle.getModifiedDate();
+
+		Assert.assertNotNull(modifiedDate);
+		Assert.assertEquals(
+			modifiedDate.getTime(), jsonObject.getLong("modifiedDate"));
+
+		Assert.assertEquals(
+			WorkflowConstants.STATUS_DRAFT, journalArticle.getStatus());
+
+		Assert.assertEquals("1", jsonObject.getString("version"));
+	}
+
+	@Test
+	public void testServeResourceWithErrors() throws Exception {
+		JSONObject jsonObject = _serveResource(
+			_getMockLiferayResourceRequest());
+
+		Assert.assertTrue(jsonObject.getBoolean("success"));
+
+		JournalArticle journalArticle =
+			_journalArticleLocalService.fetchArticle(
+				_group.getGroupId(), jsonObject.getString("articleId"));
+
+		jsonObject = _serveResource(
+			_getMockLiferayResourceRequest(
+				journalArticle.getArticleId(), Collections.emptyMap()));
+
+		Assert.assertFalse(jsonObject.getBoolean("success"));
+		Assert.assertEquals(
+			"You must define a friendly URL for the default language.",
```

Comment:
@DiegoHu97, it would be better to use the language key here. We can do it together